src/rgw/rgw_sal_motr.cc

```diff
@@ -1729,7 +1729,7 @@ int MotrObject::MotrDeleteOp::delete_obj(const DoutPrefixProvider* dpp, optional
       {
         // delete object permanently.
         result.version_id = ent.key.instance;
-        if (source->category == RGWObjCategory::MultiMeta)
+        if (ent.meta.category == RGWObjCategory::MultiMeta)
```

Comment:
why this change, source->category not reflecting proper one ? 
I can see ent.meta.category getting set in MotrMultipartUpload::init(), if this ent.meta.category reference getting used everywhere, then its fine.

Reply:
Yes, `ent.meta.category` reference is used to store the object category during multipart object PUT. 
In the delete_obj() function, `source->category` returns None as it points to the MotrObject::MotrDeleteOp. In the same function, there is a call for get_bucket_dir_ent() function to fetch the object entry which gives us all the object data into ent structure. Then we can use the same to check the category by doing `ent.meta.category`

Reply:
ok